src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/StringBuilderAppendShouldNotTakeSubstring.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeQuality.Analyzers.ApiDesignGuidelines
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class StringBuilderAppendShouldNotTakeSubstring : DiagnosticAnalyzer
+    {
+        internal const string RuleIdOneParameterId = "_1638_Substring1";
```

Comment:
>_1638_Substring1 [](start = 54, length = 16)

Can you file a tracking bug with all the TODOs in your PR?

1. Finalize resource strings
2. Finalize rule IDs
3. File Documentation bug for new rule and add it to the descriptor here.